pagopa/io-app
# feat: [IOCOM-435] Add polling for download F24 attachments

## Short description
This PR adds polling for download F24 attachments.
When the app requests the download of an F24 attachment, if the backend responds with a 503 code with the retry-after header value set, the app will wait for the time specified in the header before making the next request.

### Preview

https://github.com/pagopa/io-app/assets/29163287/29241e5b-89a3-42c1-a1b1-779d8c5b3d4a

## List of changes proposed in this pull request
- Added polling management.
- Renamed the downloadAttachment file to handleDownloadAttachment
- Renamed the clearAttachments file to handleClearAttachments
- Extracted the generator functions that initialize requests into a new file

## How to test
Generate a generic message with attachments and a PN message with attachments and F24 using `io-dev-api-server`.
In case of a 503 and retry-after, check that the requests are handled correctly: the app should wait for _n_ seconds before making the next request
